### ISSUE TITLE
Strange errors with JIT enabled.

### ISSUE THREAD

**Issue by CapsAdmin** [raptorjit/raptorjit#67]
It's hard to reproduce as these errors go away when attempting to debug. I'm running Linux.

**Comment by lukego** [comment]
Any details?

**Comment by CapsAdmin** [comment]
Not really, I was hoping for some pointers as to which commit to try or something. The codebase is big and it seems to be happening here and there randomly.

**Comment by lukego** [comment]
I think the first step is to create a reproducible test case. Even if "reproducible" means "crashes one time out of a thousand."

**Comment by lukego** [comment]
Overall the master branch should be basically stable-ish. It has not been tested very much beyond the test/benchmark suite, and RaptorJIT inherits the GC64 mode from LuaJIT which is a relatively new feature and likely contains bugs (like https://github.com/raptorjit/raptorjit/pull/47#issuecomment-287644538).

I plan to make a Snabb port to RaptorJIT soon and that will expose it to a much more intensive test suite.

**Comment by CapsAdmin** [comment]
There's an error that happens every time but I think it would be easier for me to roll back commit by commit until it disappears.

**edit:** So GC64 is enabled by default? Maybe that's the reason.

**Comment by lukego** [comment]
GC64 is enabled always. I "burned the ships" and deleted the non-GC64 code. Goal is for all the code to work properly and that is easier when there is less of it.

I would recommend `git bisect` to search for a relevant commit. Be warned though that some commits are very large e.g. removal of non-GC64.

**Comment by lukego** [comment]
There is a test suite in the `raptorjit/testsuite` directory. Maybe add your case there?

**Comment by lukego** [comment]
@CapsAdmin Did you get anywhere with this? Maybe have a reproducible bug test case?

**Comment by CapsAdmin** [comment]
Steps to reproduce this so far (no minimal case yet sorry)
```
git clone https://github.com/CapsAdmin/goluwa --depth 1 && cd goluwa
./goluwa build luajit_forks raptorjit
```

`./goluwa client branch lukego-master_52compat` normal
`./goluwa client branch lukego-master_debug-assert_52compat debug` debug symbols with gdb
`./goluwa client branch lukego-master_debug_52compat debug` debug symbols and assertions enabled with gdb
